tempodb/search/streaming_search_block.go

```diff
@@ -112,6 +86,10 @@ func (s *StreamingSearchBlock) Search(ctx context.Context, p Pipeline, sr *Resul
 			return nil
 		}
 
+		if r.Length == 0 {
+			continue
+		}
```

Comment:
Tempted to say "how did it work before", haven't we always had zero length records in the appender 🤔